Importation des packages

In [8]:
import pandas as pd
import requests

Paramétrage de l'API

In [9]:
BASE_URL_API = "https://apidf-preprod.cerema.fr"
#code_insee, nom =  "59512", "Roubaix" #"75015", "Paris"
code_insee = "78220"
url = BASE_URL_API + f"/indicateurs/dv3f/communes/annuel/{code_insee}"
# %%
print(url)

https://apidf-preprod.cerema.fr/indicateurs/dv3f/communes/annuel/78220


In [10]:
# %%
params = {
    'code_commune': '75056',  # Exemple de code commune pour Paris (75056)
    'date_min': '2022-01-01', # Transactions depuis 2022
    'date_max': '2023-01-01', # Transactions jusqu'en 2023
    'type': 'Vente',           # Type de transaction (Vente, etc.)
    'limit': 1000              # Limite de résultats
}
# %%

In [11]:
response = requests.get(url)#, params=params)
# Vérification du succès de la requête
if response.status_code == 200:
    data = response.json()  # On récupère les données au format JSON
else:
    print(f"Erreur : {response.status_code}")
# %%
print(type(data))
print(data.keys())

<class 'dict'>
dict_keys(['count', 'next', 'previous', 'results'])


Data est un dictionnaire ; on va transformer son entrée "results" en un dataframe python

In [12]:
# Conversion en DataFrame pandas
df = pd.DataFrame.from_dict(data["results"])
print(df.head())
#print(df.columns)

  annee codgeo              libgeo  nbtrans_cod1  valeurfonc_sum_cod1  \
0  2010  78220  Les Essarts-le-Roi           110          29119176.00   
1  2011  78220  Les Essarts-le-Roi           117          35915128.11   
2  2012  78220  Les Essarts-le-Roi            92          24185911.92   
3  2013  78220  Les Essarts-le-Roi            75          21053459.00   
4  2014  78220  Les Essarts-le-Roi            91          25406500.00   

   nbtrans_cod2  valeurfonc_sum_cod2  nbtrans_cod11  valeurfonc_sum_cod11  \
0            13           8231455.50             64           21563525.00   
1             9                  NaN             63           21171328.11   
2            11           4442819.93             45           14792260.98   
3             2                  NaN             34           10975843.00   
4             1                  NaN             53           16275400.00   

   nbtrans_cod111  ...  nbtrans_cod1311  valeurfonc_sum_cod1311  \
0              64  ...         

In [13]:
df.head(1)

annee codgeo              libgeo  nbtrans_cod1  valeurfonc_sum_cod1  \
0  2010  78220  Les Essarts-le-Roi           110           29119176.0   

   nbtrans_cod2  valeurfonc_sum_cod2  nbtrans_cod11  valeurfonc_sum_cod11  \
0            13            8231455.5             64            21563525.0   

   nbtrans_cod111  ...  nbtrans_cod1311  valeurfonc_sum_cod1311  \
0              64  ...                4                    None   

   nbtrans_cod14  valeurfonc_sum_cod14  nbtrans_cod142  valeurfonc_sum_cod142  \
0              0                   NaN               0                   None   

   nbtrans_cod143  valeurfonc_sum_cod143  nbtrans_cod149  \
0               0                    NaN               0   

   valeurfonc_sum_cod149  
0                   None  

[1 rows x 617 columns]

Tête très bizarre des données : vu les montants et la variable "nombre de transactions", j'ai l'impression qu'on accède plutôt à un "résumé des données DVF, où les données sont agrégées
De plus, bizarre d'avoir 617 colonnes ; et on voit dans la première colonne qu'ici on a plutôt 1ligne = une année !! pas du tout ce qu'on veut
Nous on veut une ligne = une transaction

Bref, je tente d'accéder à la base via l'API du gouvernement et pas via celle du CEREMA

In [22]:
BASE_URL_API_GOUV = " https://www.data.gouv.fr/api/1"
#code_insee, nom =  "59512", "Roubaix" #"75015", "Paris"
code_insee = "78220"
url_gouv = BASE_URL_API_GOUV + f"/dvf_opendata/geomutations/{code_insee}"
# %%
print(url_gouv)

 https://www.data.gouv.fr/api/1/dvf_opendata/geomutations/78220


In [23]:
response = requests.get(url_gouv)#, params=params)
# Vérification du succès de la requête
if response.status_code == 200:
    data_essai = response.json()  # On récupère les données au format JSON
else:
    print(f"Erreur : {response.status_code}")
# %%
print(data_essai)

Erreur : 404


NameError: name 'data_essai' is not defined